node/src/actors/chain_manager/handlers.rs

```diff
@@ -1100,10 +1115,22 @@ impl Handler<PeersBeacons> for ChainManager {
 
                         // If we are synced, it does not matter what blocks have been consolidated
                         // by our outbound peers, we will stay synced until the next superblock
-                        // vote. We will remove those that are different from the last consensus or
-                        // peers that did not send any beacon
+                        // vote. The only exception is if there are a block consensus (more than 2/3)
+                        // and we didn't consolidate any block in the last epoch, because it could
+                        // be that we lost a candidate by any reason
+                        if is_there_block_consensus
+                            && consensus_beacon.checkpoint == current_epoch - 1
+                            && our_beacon.checkpoint != current_epoch - 1
+                        {
+                            self.update_state_machine(StateMachine::WaitingConsensus);
```

Comment:
Not sure if it's safe to move to WaitingConsensus without reverting the storage.

Reply:
This corner case is when you miss one block consolidation (by any reason), and your outbounds no. So in this specific case, you are only one block behind, you would not need to revert your chain

Reply:
You assume that you will be able to synchronize to a chain that includes that block, but the network may revert and that block may no longer exist, in which case you will get a "your chain is probably forked" message. And in that case the node will revert. So I guess it's not a problem, but I'm still a bit unsure.

Reply:
It is possible to test this somehow?

Reply:
After discussing it with @lrubiorod and @tmpolaczyk , it should not be a problem because those nodes will recover after they receive a different superblock.